cmd/create/ocmrole/cmd.go

```diff
@@ -0,0 +1,391 @@
+		reporter.Infof("AWS credentials are valid!")
+	}
+
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Unable to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && (!cmd.Flags().Changed("mode")) {
+		interactive.Enable()
+	}
+
+	if reporter.IsTerminal() {
+		reporter.Infof("Creating ocm role")
+	}
+
+	prefix := args.prefix
+	if interactive.Enabled() {
+		prefix, err = interactive.GetString(interactive.Input{
+			Question: "Role prefix",
+			Help:     cmd.Flags().Lookup("prefix").Usage,
+			Default:  prefix,
+			Required: true,
+			Validators: []interactive.Validator{
+				interactive.RegExp(`[\w+=,.@-]+`),
+				interactive.MaxLength(32),
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role prefix: %s", err)
+			os.Exit(1)
+		}
+	}
+	if len(prefix) > 32 {
+		reporter.Errorf("Expected a prefix with no more than 32 characters")
+		os.Exit(1)
+	}
+	if !aws.RoleNameRE.MatchString(prefix) {
+		reporter.Errorf("Expected a valid role prefix matching %s", aws.RoleNameRE.String())
+		os.Exit(1)
+	}
+
+	permissionsBoundary := args.permissionsBoundary
+	if interactive.Enabled() {
+		permissionsBoundary, err = interactive.GetString(interactive.Input{
+			Question: "Permissions boundary ARN",
+			Help:     cmd.Flags().Lookup("permissions-boundary").Usage,
+			Default:  permissionsBoundary,
+			Validators: []interactive.Validator{
+				aws.ARNValidator,
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid policy ARN for permissions boundary: %s", err)
+			os.Exit(1)
+		}
+	}
+	if permissionsBoundary != "" {
+		_, err := arn.Parse(permissionsBoundary)
+		if err != nil {
+			reporter.Errorf("Expected a valid policy ARN for permissions boundary: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "Role creation mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  aws.ModeAuto,
+			Options:  aws.Modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role creation mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	// Get current OCM account:
+	orgID, err := ocmClient.GetCurrentOrganization()
+	if err != nil {
+		reporter.Errorf("Failed to get organization account: %s", err)
+		os.Exit(1)
+	}
+
+	switch mode {
+	case aws.ModeAuto:
+		reporter.Infof("Creating role using '%s'", creator.ARN)
+		roleARN, err := createRoles(reporter, awsClient, prefix, permissionsBoundary, creator.AccountID,
+			orgID, env)
+		if err != nil {
+			reporter.Errorf("There was an error creating the ocm role: %s", err)
+			ocmClient.LogEvent("ROSACreateOCMRoleModeAuto", map[string]string{
+				ocm.Response: ocm.Failure,
+			})
+			os.Exit(1)
+		}
+		ocmClient.LogEvent("ROSACreateOCMRoleModeAuto", map[string]string{
+			ocm.Response: ocm.Success,
+		})
+		err = ocmrole.Cmd.RunE(cmd, []string{roleARN})
```

Comment:
`ocmrole` here is confusing, you should use a package alias (e.g. `linkocmrole`)